wallet/chains/pchain/sender.go

```diff
@@ -3,27 +3,75 @@
 package pchain
 
 import (
+	"errors"
 	"fmt"
+	"time"
 
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
 	"github.com/ava-labs/avalanchego/wallet/subnet/primary"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
 
+	"github.com/ava-labs/avalanche-tooling-sdk-go/multisig"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/utils"
 	"github.com/ava-labs/avalanche-tooling-sdk-go/wallet/types"
-
-	avagoTxs "github.com/ava-labs/avalanchego/vms/platformvm/txs"
 )
 
+// sendWithRetry submits a transaction to the network with retry logic
+func sendWithRetry(wallet *primary.Wallet, tx *txs.Tx) error {
+	const (
+		repeats             = 3
+		sleepBetweenRepeats = 2 * time.Second
+	)
+	var issueTxErr error
+	for i := 0; i < repeats; i++ {
+		ctx, cancel := utils.GetAPILargeContext()
+		defer cancel()
+		options := []common.Option{common.WithContext(ctx)}
+		// TODO: split error checking and recovery between issuing and waiting for status
+		issueTxErr = wallet.P().IssueTx(tx, options...)
+		if issueTxErr == nil {
+			break
+		}
+		if ctx.Err() != nil {
+			issueTxErr = fmt.Errorf("timeout issuing/verifying tx with ID %s: %w", tx.ID(), issueTxErr)
+		} else {
+			issueTxErr = fmt.Errorf("error issuing tx with ID %s: %w", tx.ID(), issueTxErr)
+		}
+		time.Sleep(sleepBetweenRepeats)
+	}
+	if issueTxErr != nil {
+		return fmt.Errorf("issue tx error %w", issueTxErr)
+	}
+	return nil
+}
+
 // SendTx sends P-Chain transactions
```

Comment:
I think we should add to the comment here the specifics of the retry logic to be more clear

Reply:
added